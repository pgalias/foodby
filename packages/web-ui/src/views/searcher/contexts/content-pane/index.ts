export {
  ContentPaneProvider,
  useChangeCurrentPaneState,
  useCurrentPaneState,
  ContentPageState,
} from './provider';
