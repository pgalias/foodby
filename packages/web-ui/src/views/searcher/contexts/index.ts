export * from './content-pane';
export * from './filters';
