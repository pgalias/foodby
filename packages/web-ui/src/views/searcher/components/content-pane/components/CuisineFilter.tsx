import React, { FC } from 'react';
import { Card, SelectableTag } from '@foodby/common-components-ui';
import clsx from 'clsx';
import { Types, useFilterDispatch, useFilters } from '../../../contexts';
import { useCuisineTypes } from '../../../hooks';

export const CuisineFilter: FC = () => {
  const { data } = useCuisineTypes();
  const { cuisineTypes } = useFilters();
  const dispatch = useFilterDispatch();

  const onSelect = (value: string, isSelected: boolean) => {
    const currentCuisines = isSelected
      ? [...cuisineTypes, value]
      : cuisineTypes.filter((cuisine) => cuisine !== value);

    dispatch({
      type: Types.SET_CUISINE_TYPES,
      payload: currentCuisines,
    });
  };

  return (
    <Card className={clsx('mx-auto', 'h-48', 'center-y')}>
      <h3 className={clsx('text-sm text-coolGray-400', 'mb-4 -mt-5', 'block')}>
        Cuisines
      </h3>
      <div className={clsx('grid grid-cols-5 gap-2')}>
        {data?.map((cuisine) => (
          <SelectableTag
            key={cuisine}
            name={cuisine}
            value={cuisine}
            isSelected={cuisineTypes.includes(cuisine)}
            onSelect={onSelect}
          />
        ))}
      </div>
    </Card>
  );
};
