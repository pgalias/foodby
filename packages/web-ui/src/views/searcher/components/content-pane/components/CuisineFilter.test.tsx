import * as React from 'react';
import { render, screen } from '@testing-library/react';
import userEvents from '@testing-library/user-event';
import { CuisineFilter } from './CuisineFilter';
import { Types } from '../../../contexts';

const mockDispatch = jest.fn();

jest.mock('../../../contexts', () => ({
  ...(jest.requireActual('../../../contexts') as Record<string, unknown>),
  useFilterDispatch: () => mockDispatch,
  useFilters: jest.fn(() => ({
    cuisineTypes: ['french', 'bulgarian'],
  })),
}));

jest.mock('../../../hooks', () => ({
  ...(jest.requireActual('../../../hooks') as Record<string, unknown>),
  useCuisineTypes: jest.fn(() => ({
    data: ['french', 'italian', 'polish', 'belgian', 'bulgarian', 'japanese'],
  })),
}));

describe('CuisineFilter', () => {
  test('should render a card and selectable tags components', () => {
    const { asFragment } = render(<CuisineFilter />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should select cuisine tags based on the value of cuisine types from the filters store', () => {
    render(<CuisineFilter />);
    expect(screen.queryAllByRole('checkbox', { checked: true })).toHaveLength(
      2,
    );
  });

  test('should dispatch an action with collection of cuisines extended with clicked one when it was not selected previously', () => {
    render(<CuisineFilter />);
    userEvents.click(screen.getByRole('checkbox', { name: 'japanese' }));
    expect(mockDispatch).toHaveBeenCalledWith({
      type: Types.SET_CUISINE_TYPES,
      payload: ['french', 'bulgarian', 'japanese'],
    });
  });

  test('should dispatch an action with a collection of cuisines reduced by clicked one when it was selected previously', () => {
    render(<CuisineFilter />);
    userEvents.click(screen.getByRole('checkbox', { name: 'french' }));
    expect(mockDispatch).toHaveBeenCalledWith({
      type: Types.SET_CUISINE_TYPES,
      payload: ['bulgarian'],
    });
  });
});
