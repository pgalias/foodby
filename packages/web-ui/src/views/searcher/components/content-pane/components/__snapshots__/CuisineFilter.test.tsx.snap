// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CuisineFilter should render a card and selectable tags components 1`] = `
<DocumentFragment>
  <section
    class="fb-card w-2/5 p-8 bg-white text-blueGray-600 rounded-lg overflow-hidden mx-auto h-48 center-y"
  >
    <button
      aria-checked="true"
      class="fb-selectable-tag rounded-lg py-1 px-2 text-xs outline-none transition-colors bg-red-400 hover:bg-red-500 focus:bg-red-500 text-blueGray-100"
      role="checkbox"
      type="button"
    >
      french
    </button>
    <button
      aria-checked="false"
      class="fb-selectable-tag rounded-lg py-1 px-2 text-xs outline-none transition-colors bg-coolGray-200 text-blueGray-600 hover:bg-coolGray-300 focus:bg-coolGray-300"
      role="checkbox"
      type="button"
    >
      italian
    </button>
    <button
      aria-checked="false"
      class="fb-selectable-tag rounded-lg py-1 px-2 text-xs outline-none transition-colors bg-coolGray-200 text-blueGray-600 hover:bg-coolGray-300 focus:bg-coolGray-300"
      role="checkbox"
      type="button"
    >
      polish
    </button>
    <button
      aria-checked="false"
      class="fb-selectable-tag rounded-lg py-1 px-2 text-xs outline-none transition-colors bg-coolGray-200 text-blueGray-600 hover:bg-coolGray-300 focus:bg-coolGray-300"
      role="checkbox"
      type="button"
    >
      belgian
    </button>
    <button
      aria-checked="true"
      class="fb-selectable-tag rounded-lg py-1 px-2 text-xs outline-none transition-colors bg-red-400 hover:bg-red-500 focus:bg-red-500 text-blueGray-100"
      role="checkbox"
      type="button"
    >
      bulgarian
    </button>
    <button
      aria-checked="false"
      class="fb-selectable-tag rounded-lg py-1 px-2 text-xs outline-none transition-colors bg-coolGray-200 text-blueGray-600 hover:bg-coolGray-300 focus:bg-coolGray-300"
      role="checkbox"
      type="button"
    >
      japanese
    </button>
  </section>
</DocumentFragment>
`;
